test/widgets/recent_dm_conversations_test.dart

```diff
@@ -206,39 +217,95 @@ void main() {
+          });
+
+          testWidgets('non-bot recipient -> shows no bot icon', (tester) async {
+            await checkRecipient(tester, isBot: false, looksBot: false);
+          });
         });
 
-        testWidgets('short name takes one line', (WidgetTester tester) async {
-          final user = eg.user(userId: 1, fullName: 'Short name');
-          final message = eg.dmMessage(from: eg.selfUser, to: [user]);
-          await setupPage(tester, users: [user], dmMessages: [message]);
-          checkTitle(tester, user.fullName, 1);
+        group('short name takes one line', () {
+          Future<void> checkRecipient(WidgetTester tester, {
+            required bool isBot,
+            required bool looksBot
+          }) async {
+            final user = eg.user(userId: 1, fullName: 'Short name', isBot: isBot);
+            final message = eg.dmMessage(from: eg.selfUser, to: [user]);
+            await setupPage(tester, users: [user], dmMessages: [message]);
+            checkTitle(tester, user.fullName, 1);
+            checkBotIcon(count: looksBot ? 1 : 0);
+          }
+
+          testWidgets('bot recipient -> shows bot icon', (tester) async {
+            await checkRecipient(tester, isBot: true, looksBot: true);
+          });
+
+          testWidgets('non-bot recipient -> shows no bot icon', (tester) async {
+            await checkRecipient(tester, isBot: false, looksBot: false);
+          });
         });
 
-        testWidgets('very long name takes two lines (must be ellipsized)', (WidgetTester tester) async {
-          final user = eg.user(userId: 1, fullName: 'Long name long name long name long name long name long name long name long name long name long name long name long name long name long name long name long name long name long name long name long name long name long name');
-          final message = eg.dmMessage(from: eg.selfUser, to: [user]);
-          await setupPage(tester, users: [user], dmMessages: [message]);
-          checkTitle(tester, user.fullName, 2);
+        group('very long name takes two lines (must be ellipsized)', () {
+          Future<void> checkRecipient(WidgetTester tester, {
+            required bool isBot,
+            required bool looksBot
+          }) async {
+            final user = eg.user(userId: 1, isBot: isBot, fullName: 'Long name long name long name long name long name long name long name long name long name long name long name long name long name long name long name long name long name long name long name long name long name long name');
+            final message = eg.dmMessage(from: eg.selfUser, to: [user]);
+            await setupPage(tester, users: [user], dmMessages: [message]);
+            checkTitle(tester, user.fullName, 2);
+            checkBotIcon(count: looksBot ? 1 : 0);
+          }
+
+          testWidgets('bot recipient -> shows no bot icon', (WidgetTester tester) async {
+            await checkRecipient(tester, isBot: true, looksBot: false);
+          });
```

Comment:
We would like the bot icon to appear even when the name is long; see Alya's comment: https://github.com/zulip/zulip-flutter/pull/764#issuecomment-2207386553

If that's hard to do and has to be left as a TODO, that might be OK, but we shouldn't have a test that confirms the undesired behavior. 🙂